### ISSUE TITLE
Tapioca generating empty RBI file for some gems

### ISSUE THREAD

**Issue by iMacTia** [Shopify/tapioca#1212]
We're observing some strange behaviour from `tapioca` where it generates empty RBIs for some of our gems.
The surprising part is that if I run `srb rbi gems` instead (which I understand is deprecated in favour of `tapioca`), these RBI files are correctly generated 🤔.

As an example, we have `neatjson` in our Gemfile, which adds a `neat_generate` method to `JSON` (see [code](https://github.com/Phrogz/NeatJSON/blob/master/lib/neatjson.rb#L38)), but the following file gets generated by `tapioca`:

```ruby
# typed: true

# DO NOT EDIT MANUALLY
# This is an autogenerated file for types exported from the `neatjson` gem.
# Please instead update this file by running `bin/tapioca gem neatjson`.

# THIS IS AN EMPTY RBI FILE.
# see https://github.com/Shopify/tapioca#manually-requiring-parts-of-a-gem
```

I've already tried explicitly requiring `neatjson` in `require.rb`, but that doesn't make any difference.
Again, `sorbet` somehow is able to generate this gem's RBI file correctly, so I'm not really sure what's the difference or what we could be doing wrong.

**Comment by paracycle** [comment]
> I've already tried explicitly requiring neatjson in require.rb, but that doesn't make any difference.

Did you run `bin/tapioca gem neatjson` after making that change? None of the normal commands would pick up the change and regenerate the gem RBI for you.

**Comment by iMacTia** [comment]
It seemed easier than individually requiring each gem 😅 it allows me not to worry about missing requires in this file.
Is there any real functional side-effect apart from loading too much stuff (and potentially making the command slower)?

**Comment by paracycle** [comment]
Ok, I have a fix for this: https://github.com/Shopify/tapioca/pull/1213

**Comment by iMacTia** [comment]
Thank you for the super-quick fix and for clarifying those points about the require.rob file @paracycle 🙏!

Tomorrow morning I'll try your fix in our app and provide feedback 🙌

**Comment by paracycle** [comment]
Yes, that's the expected output. It is harder to figure out which value constants are coming from which gem, so we'd been conservative in filtering those out, so it is expected that those constants on JSON are in the output.
